temporal-sdk/src/test/java/io/temporal/internal/sync/QueryDispatcherTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.sync;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.temporal.api.common.v1.Payloads;
+import io.temporal.common.converter.DataConverter;
+import io.temporal.common.interceptors.WorkflowInboundCallsInterceptor;
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.junit.Test;
+
+public class QueryDispatcherTest {
+
+  private QueryDispatcher initializeDispatcher() {
```

Comment:
Might be a good idea to pass a list of supported queries here, e.g.
```
initializeDispatcher(String... queries) {
...
```
This would make tests a bit more readable.